libs/universal-dapp.js

```diff
@@ -0,0 +1,333 @@
+                args.appendFunctions(result.createdAddress);
+            } else if (self.options.vm){
+                var outputObj = fun.unpackOutput('0x' + result.vm.return.toString('hex'));
+                $result.replaceWith( getOutput( getReturnOutput( outputObj ), getGasUsedOutput( result.vm ) ) );
+            } else if (args.abi.constant && !isConstructor) {
+                $result.replaceWith( getOutput( getReturnOutput( result ) ) );
+            } else {
+                
+                function tryTillResponse (txhash, done) {
+                    web3.eth.getTransactionReceipt(result, testResult );
+
+                    function testResult (err, address) {
+                        if (!err && !address) {
+                            setTimeout( function(){ tryTillResponse(txhash, done) }, 500)
+                        } else done( err, address )
+                    }
+
+                }
+                tryTillResponse( result, function(err, result) {
+                    if (isConstructor) {
+                        $result.html('');
+                        args.appendFunctions(result.contractAddress);
+                    } else $result.replaceWith( getOutput( getReturnOutput( result ), getGasUsedOutput( result ) ) );
+                })
+            
+            }
+        });
+    }
+
+    var button = $('<button />')
+        .addClass( 'call' )
+        .text(args.bytecode ? 'Create' : fun.displayName())
+        .click( handleCallButtonClick );
+
+    if (lookupOnly && !inputs.length) {
+        handleCallButtonClick();
+    }
+
+    var $contractProperty = $('<div class="contractProperty"/>');
+    $contractProperty
+        .toggleClass( 'constant', !isConstructor && args.abi.constant )
+        .toggleClass( 'hasArgs', args.abi.inputs.length > 0)
+        .toggleClass( 'constructor', isConstructor)
+        .append(button)
+        .append( (lookupOnly && !inputs.length) ? $outputOverride : inputField );
+    return $contractProperty.append(outputSpan);
+}
+
+UniversalDApp.prototype.clickNewContract = function ( self, $contract, contract ) {
+    $contract.append( self.getInstanceInterface(contract) );
+}
+
+UniversalDApp.prototype.clickContractAt = function ( self, $contract, contract ) {
+    var address = prompt( "What Address is this contract at in the Blockchain? ie: '0xdeadbeaf...'" )   
+    self.getInstanceInterface(contract, address, $contract );
+}
+
+UniversalDApp.prototype.runTx = function( data, args, cb) {
+    var to = args.address;
+    var constant = args.abi.constant;
+    var isConstructor = args.bytecode !== undefined;
+    
+    if (!this.vm) {
+        if (constant && !isConstructor) {
+            var func = web3.eth.contract( [args.abi] ).at( to );
+            func[args.abi.name].call( cb );
+        } else {
+            web3.eth.sendTransaction({
+                from: web3.eth.accounts[0],
+                to: to,
+                data: data,
+                gas: 1000000
```

Comment:
Ah, the dreaded gas parameter. Can we use gas estimation to find a better value?